src/test/java/itest/ReportJwtDownloadIT.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright The Cryostat Authors
+ *
+ * The Universal Permissive License (UPL), Version 1.0
+ *
+ * Subject to the condition set forth below, permission is hereby granted to any
+ * person obtaining a copy of this software, associated documentation and/or data
+ * (collectively the "Software"), free of charge and under any and all copyright
+ * rights in the Software, and any and all patent rights owned or freely
+ * licensable by each licensor hereunder covering either (i) the unmodified
+ * Software as contributed to or provided by such licensor, or (ii) the Larger
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package itest;
+
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.json.JsonObject;
+import itest.bases.JwtAssetsSelfTest;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ReportJwtDownloadIT extends JwtAssetsSelfTest {
```

Comment:
I ran `mvn clean verify` a few times locally and saw one itest failure each time. Is this related to #666?
```
[INFO] Running itest.ReportJwtDownloadIT
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 10.988 s <<< FAILURE! - in itest.ReportJwtDownloadIT
[ERROR] testDownloadRecordingUsingJwt  Time elapsed: 10.985 s  <<< ERROR!
java.util.concurrent.ExecutionException: java.lang.Exception: HTTP 500
	at itest.ReportJwtDownloadIT.testDownloadRecordingUsingJwt(ReportJwtDownloadIT.java:68)
Caused by: java.lang.Exception: HTTP 500
```
https://gist.github.com/jan-law/64b3b3adbd94ecbb886253cd63eb9ed8

```
[INFO] Running itest.ReportIT
HTTP 404: Not Found
HTTP 500: Internal Server Error
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.164 s <<< FAILURE! - in itest.ReportIT
[ERROR] testGetReportShouldSendFile  Time elapsed: 1.152 s  <<< ERROR!
java.util.concurrent.ExecutionException: io.vertx.ext.web.handler.impl.HttpStatusException: Internal Server Error
	at itest.ReportIT.testGetReportShouldSendFile(ReportIT.java:160)
Caused by: io.vertx.ext.web.handler.impl.HttpStatusException: Internal Server Error
```
https://gist.github.com/jan-law/a9a6ba755da6fe91a6f55a6d5a144c7b

```
[INFO] Running itest.RecordingWorkflowIT
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 6.49 s <<< FAILURE! - in itest.RecordingWorkflowIT
[ERROR] testWorkflow  Time elapsed: 6.49 s  <<< ERROR!
java.util.concurrent.ExecutionException: java.lang.Exception: HTTP 500
	at itest.RecordingWorkflowIT.testWorkflow(RecordingWorkflowIT.java:231)
Caused by: java.lang.Exception: HTTP 500
```
https://gist.github.com/jan-law/7a7d22c7e28196effbe58b80dcc33428

Reply:
I would guess it is related, but it's hard to be sure from those logs. Next time it occurs, could you also upload the `target/cryostat-itest-*.log`? That's the log file taken from the Cryostat process inside the container, so it's more informative than the integration test client side logs that are printed out - it should contain a stack trace for those 500s, for example.

Reply:
Here's the workflow itest failure log. There's an `EmptyRecordingException` and a `ReportGenerationException` that triggered a 500:
https://gist.github.com/jan-law/0acb00c5ad349d4ce532fdcd11e9ae18

Reply:
The `EmptyRecordingException` is related to #743 and the `ReportGenerationException` does look just like #666